stl/inc/format

```diff
@@ -2370,24 +2370,48 @@ _NODISCARD _OutputIt _Fmt_write(
     auto _Separators       = 0;
     string _Groups;
 
-    if ((_Specs._Alt || _Specs._Localized) && _Is_finite) {
-        for (auto _It = _Buffer_start; _It < _Result.ptr; ++_It) {
-            if (*_It == '.') {
-                _Radix_point = _It;
-            } else if (*_It == _Exponent) {
-                _Exponent_start = _It;
+    if (_Is_finite) {
+        if (_Specs._Alt || _Specs._Localized) {
+            for (auto _It = _Buffer_start; _It < _Result.ptr; ++_It) {
+                if (*_It == '.') {
+                    _Radix_point = _It;
+                } else if (*_It == _Exponent) {
+                    _Exponent_start = _It;
+                }
             }
-        }
-        _Integral_end = (_STD min) (_Radix_point, _Exponent_start);
+            _Integral_end = (_STD min) (_Radix_point, _Exponent_start);
 
-        if (_Specs._Alt) {
-            if (_Radix_point == _Result.ptr) {
+            if (_Specs._Alt && _Radix_point == _Result.ptr) {
                 // TRANSITION, decimal point may be wider
                 ++_Width;
                 _Append_decimal = true;
             }
 
-            if (_Specs._Type == 'g' || _Specs._Type == 'G') {
+            if (_Specs._Localized) {
+                _Groups     = _STD use_facet<numpunct<_CharT>>(_Locale._Get()).grouping();
+                _Separators = _Count_separators(_Integral_end - _Buffer_start, _Groups);
+            }
+        }
+
+        switch (_Specs._Type) {

```

Comment:
I think this should be `switch`ed on `_Format`.

Reply:
A few lines above we have:
```cpp
 switch (_Specs._Type) {
    case '\0':
        break;
    case 'A':
        _To_upper = true;
        [[fallthrough]];
    case 'a':
        _Format   = chars_format::hex;
        _Exponent = 'p';
        break;
```

So I believe this is the right choice